<a href="https://colab.research.google.com/github/andrea0026/proyecto2/blob/main/Proyecto2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('/content/drive/MyDrive/Data_science/Machine learning/archivos/water_potability.csv')
df.head()

ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0       NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1  3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2  8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3  8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4  9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0

Primera opción: conjunto de datos 1

1) Fuente de datos: https://www.kaggle.com/datasets/adityakadiwal/water-potability

2) Breve descripción de datos: Este data set muestra 9 caracteristicas sobre la calidad del agua, a partir de estas se determina si el agua es potable o no.

3) ¿Cuál es el objetivo? la colunma objetivo es Potability

4) ¿Este es un problema de clasificación o regresión? es un problema de clasificacion

5) ¿Cuántas características hay? 10 colunmas, (9 caracteristicas y 1 objetivo)

6) ¿Cuántas filas de datos hay? 3275 filas

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos? Se debe iniciar con la correcta imputacion de valores faltantes, ademas el dataset es adecuado para probar diferentes tipos de modelos de clasificacion. 

In [3]:
df = pd.read_csv('/content/drive/MyDrive/Data_science/Machine learning/archivos/parkinsons.data')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  MDVP:Jitter(%)  \
0  phon_R01_S01_1      119.992       157.302        74.997         0.00784   
1  phon_R01_S01_2      122.400       148.650       113.819         0.00968   
2  phon_R01_S01_3      116.682       131.111       111.555         0.01050   
3  phon_R01_S01_4      116.676       137.871       111.366         0.00997   
4  phon_R01_S01_5      116.014       141.781       110.655         0.01284   

   MDVP:Jitter(Abs)  MDVP:RAP  MDVP:PPQ  Jitter:DDP  MDVP:Shimmer  ...  \
0           0.00007   0.00370   0.00554     0.01109       0.04374  ...   
1           0.00008   0.00465   0.00696     0.01394       0.06134  ...   
2           0.00009   0.00544   0.00781     0.01633       0.05233  ...   
3           0.00009   0.00502   0.00698     0.01505       0.05492  ...   
4           0.00011   0.00655   0.00908     0.01966       0.06425  ...   

   Shimmer:DDA      NHR     HNR  status      RPDE       DFA   spread1  \
0      0.06545  0.02211  21.033       1  0.414783  0.815285 -4.813031   
1      0.09403  0.01929  19.085       1  0.458359  0.819521 -4.075192   
2      0.08270  0.01309  20.651       1  0.429895  0.825288 -4.443179   
3      0.08771  0.01353  20.644       1  0.434969  0.819235 -4.117501   
4      0.10470  0.01767  19.649       1  0.417356  0.823484 -3.747787   

    spread2        D2       PPE  
0  0.266482  2.301442  0.284654  
1  0.335590  2.486855  0.368674  
2  0.311173  2.342259  0.332634  
3  0.334147  2.405554  0.368975  
4  0.234513  2.332180  0.410335  

[5 rows x 24 columns]

In [5]:
df.shape

(195, 24)

Primera opción: conjunto de datos 1

1) Fuente de datos: https://archive.ics.uci.edu/ml/datasets/parkinsons

2) Breve descripción de datos: El objetivo principal de los datos es discriminar a las personas sanas de las que tienen PD, según la columna de "status", que se establece en 0 para sanos y 1 para PD.

3) ¿Cuál es el objetivo? la colunma status

4) ¿Este es un problema de clasificación o regresión? es un problema de clasificacion. 

5) ¿Cuántas características hay? 23 

6) ¿Cuántas filas de datos hay? 195

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos? Se pueden aplicar diferentes modelos para clasificacion. 